### ISSUE TITLE
Investigate and divulge effects of setting the __intra loop sleep duration__

### ISSUE THREAD

**Issue by lasarojc** [cometbft/cometbft#2675]
This configuration parameter

https://github.com/cometbft/cometbft/blob/1884beb6ce7689a61b5dcb2c70f30ef0aa46047f/config/config.go#L1135

is used to define the maximum value for a sleep inside the `gossipDataRoutine`

https://github.com/cometbft/cometbft/blob/1884beb6ce7689a61b5dcb2c70f30ef0aa46047f/internal/consensus/reactor.go#L579

Obviously if nodes sleep forever then no bandwidth is consumed in this function and if they sleep for no time then the function will burst information out as fast as possible. 
Hence intermediate values should affect the bandwidth usage and could be a knob to tweak the amount of bandwidth used.
In this issue we investigate the how varying this parameter varies the network usage.

## Setup
In a 175 node network, with 175 validators, with a topology established according to this [PR](https://github.com/cometbft/qa-infra/blob/lasarojc/vary_ILSD/script/runtests/vary_sleep_testnet_tmpl.toml), the Intra Loop Sleep Delay (ILSD) was set to  500ms, 400ms, 300ms, 250ms, 200ms, 150ms, 100ms, 50ms, 0ms (baseline) in a series of experiments.
Each experiment had 5 runs with transactions being injected by one node at 200tx/s for 90 seconds.

The following image shows that even though blocks are produced less frequently (chain height) with higher values, the throughput in terms of transactions (chain size txs) remained visually the same, while the block sizes decreased.

![image](https://github.com/cometbft/cometbft/assets/2686194/6ed3ec9e-0b57-4396-a866-6230270d19f7)

The following image shows the bandwidth usage, with clear increase for BlockPart messages.

![image](https://github.com/cometbft/cometbft/assets/2686194/916a2dca-95d7-49ee-839e-74535b93fc3d)

The following is a zoom to highlight the increase of messages of type Vote

![image](https://github.com/cometbft/cometbft/assets/2686194/021f1c58-55c8-4616-9752-6928c957202b)

The following graphs show the increase in the duplicate reception of block parts and votes.

![image](https://github.com/cometbft/cometbft/assets/2686194/2c988aaf-ecbc-47b3-a256-097fecbb3543)

## Initial conclusions

- For the same transaction (low) throughput, increasing the delay reduces network usage by allowing information about having seen a vote or block to propagate before the vote or block is sent to the node.
- 0 probably not the best default value.
- Different kinds of messages should experience different delays

## DoD
- Determine if using larger sleeps have a detrimental effect on the throughput under higher system loads.
- Determine a better default value for the parameter
- Experiment with different delays for different kinds of messages (Block parts x POL)

**Comment by lasarojc** [comment]
Repeated the experiment for 4 times the load, injected by 4 different nodes, for all the sleep values except 0ms
The following graphs show that the value greatly affects the block throughput, as expected

![image](https://github.com/cometbft/cometbft/assets/2686194/5257580c-4a5f-4f28-8a2e-2aec2b561f1f)

but also in terms of tx/s

![image](https://github.com/cometbft/cometbft/assets/2686194/8aabfbe0-74aa-40a6-9165-250066e5e743)

this likely happened because more transactions were more readily available to proposers, since the injection was distributed.

Observe that the throughput was higher even if more than 1 round was frequently needed to decide 

![image](https://github.com/cometbft/cometbft/assets/2686194/5f89e9cd-da32-4316-9c4b-7b827a7bb56f)

Duplication of blocks and votes increased faster in these runs

![image](https://github.com/cometbft/cometbft/assets/2686194/a004af68-a94d-4852-beeb-ce66a340153f)

![image](https://github.com/cometbft/cometbft/assets/2686194/99970494-c3ec-4879-ac2f-99a03fbcf6ba)

Block parts and txs bandwidth also increased, probably because larger delays implied in the mempool being full more time 

![image](https://github.com/cometbft/cometbft/assets/2686194/0bab98f4-55fa-4540-b59c-6de706f7b056)

![image](https://github.com/cometbft/cometbft/assets/2686194/90cdad9b-47a0-481d-b8a2-73a99640e1d7)

**Comment by lasarojc** [comment]
My conclusion is that it is clear that this parameter can affect the chain's usage of network and throughput and therefore it must be further investigated.

**Comment by sergio-mena** [comment]
@adizere @andynog Would it be worth turning this into a blog?
@lasarojc is there a "safe and obvious" value we can use as default for `v1` and beyond?

**Comment by lasarojc** [comment]
In order to propose a default these experiments should repeated with LE as I ran all these experiments without LE, which may exacerbate the benefits of setting this option to higher values.

**Comment by cason** [comment]
There are good indications that this approach works, we need to establish here what is a good default for v1.x

**Comment by sergio-mena** [comment]
Agree. We probably need to re-run Lásaro's experiments with delay emulation, and explain the whole thing in a blogpost. @adizere, @andynog WDYT?
